central/cve/datastoretest/datastore_sac_test.go

```diff
@@ -0,0 +1,1239 @@
+	return s.genericStore.Count(ctx, q)
+}
+
+func (s *nodeCVEDataStoreFromGenericStore) GetBatch(ctx context.Context, id []string) ([]*storage.NodeCVE, error) {
+	cves, err := s.genericStore.GetBatch(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	nodeCVEs := make([]*storage.NodeCVE, 0, len(cves))
+	for _, cve := range cves {
+		if !isNodeCVE(cve) {
+			continue
+		}
+		nodeCVEs = append(nodeCVEs, converter.ProtoCVEToNodeCVE(cve))
+	}
+	return nodeCVEs, nil
+}
+
+func (s *nodeCVEDataStoreFromGenericStore) Suppress(ctx context.Context, start *types.Timestamp, duration *types.Duration, cves ...string) error {
+	return s.genericStore.Suppress(ctx, start, duration, cves...)
+}
+
+func (s *nodeCVEDataStoreFromGenericStore) Unsuppress(ctx context.Context, cves ...string) error {
+	return s.genericStore.Unsuppress(ctx, cves...)
+}
+
+func (s *nodeCVEDataStoreFromGenericStore) EnrichNodeWithSuppressedCVEs(node *storage.Node) {
+	s.genericStore.EnrichNodeWithSuppressedCVEs(node)
+}
+
+func (s *cveDataStoreSACTestSuite) SetupSuite() {
+	var err error
+	s.dackboxTestStore, err = dackboxTestUtils.NewDackboxTestDataStore(s.T())
+	s.Require().NoError(err)
+	if features.PostgresDatastore.Enabled() {
+		pool := s.dackboxTestStore.GetPostgresPool()
+		s.imageCVEStore, err = imageCVEDataStore.GetTestPostgresDataStore(s.T(), pool)
+		s.Require().NoError(err)
+		s.nodeCVEStore, err = nodeCVEDataStore.GetTestPostgresDataStore(s.T(), pool)
+		s.Require().NoError(err)
+	} else {
+		dacky := s.dackboxTestStore.GetDackbox()
+		keyFence := s.dackboxTestStore.GetKeyFence()
+		rocksEngine := s.dackboxTestStore.GetRocksEngine()
+		bleveIndex := s.dackboxTestStore.GetBleveIndex()
+		indexQ := s.dackboxTestStore.GetIndexQ()
+		genericCVEStore, err := genericCVEDataStore.GetTestRocksBleveDataStore(s.T(), rocksEngine, bleveIndex, dacky,
+			keyFence, indexQ)
+		s.Require().NoError(err)
+		s.imageCVEStore = &imageCVEDataStoreFromGenericStore{
+			genericStore: genericCVEStore,
+		}
+		s.nodeCVEStore = &nodeCVEDataStoreFromGenericStore{
+			genericStore: genericCVEStore,
+		}
+	}
+	s.imageTestContexts = sacTestUtils.GetNamespaceScopedTestContexts(context.Background(), s.T(), resources.Image)
+	s.nodeTestContexts = sacTestUtils.GetNamespaceScopedTestContexts(context.Background(), s.T(), resources.Node)
+}
+
+func (s *cveDataStoreSACTestSuite) TearDownSuite() {
+	s.Require().NoError(s.dackboxTestStore.Cleanup())
+}
+
+func (s *cveDataStoreSACTestSuite) getCVEID(cve string) string {
+	if features.PostgresDatastore.Enabled() {
+		return cve + "#Linux"
+	}
+	return cve
+}
+
+func (s *cveDataStoreSACTestSuite) cleanImageToVulnerabilitiesGraph() {
+	s.Require().NoError(s.dackboxTestStore.CleanImageToVulnerabilitiesGraph())
+}
+
+func (s *cveDataStoreSACTestSuite) cleanNodeToVulnerabilitiesGraph() {
+	s.Require().NoError(s.dackboxTestStore.CleanNodeToVulnerabilitiesGraph())
+}
+
+type existsTestCase struct {
```

Comment:
I think we have quite some duplicated code, specifically regarding the test case definitions.

Would it make sense to combine them based on the type we are testing for (where type == single scope CVE, non-shared, shared)?

What I'm suggestion would be looking like the following:
```go
type imageCVETestcase struct {
  contextKey string
  expectExists map[string]bool
  expectFound map[string]bool
  expectedCount map[string]int
}

func createTestCases() map[string]imageCVETestcase {
  return map[string]imageCVETestcase{
    "some test case description": {
       contextKey: <some-context-key>,
       expectExists: map[string]bool{
         "CVE-ID1": true,
         "CVE-ID2": false,
       },
       // and so on....
    }
  }
}

// Within the tests, usage would look like the following:
for name, c := range createTestCases {
		s.Run(name, func() {
			testCtx := s.imageTestContexts[c.contextKey]
			exists, err := s.imageCVEStore.Exists(testCtx, cveId)
			s.NoError(err)
                        // Using the CVE-ID that's used for the test.
			s.Equal(c.expectedExists[CVE-ID], exists)
		})
}
```
 It's a bit more complex to initially create the test cases, but at the very least we won't have them duplicated for the three different scopes (single, not shared, shared) for `Exists, Get, Count`. Would make sure we use the same test cases across the different scopes, potentially making maintenance of them also easier.

WDYT about making this adjustment?